FinnUI/Sources/Components/SearchFilterTags/SearchFilterTagsView.swift

```diff
@@ -0,0 +1,255 @@
+        return cell
+    }
+}
+
+// MARK: - UICollectionViewDelegate
+
+extension SearchFilterTagsView: UICollectionViewDelegate {
+    public func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        let offset = scrollView.contentOffset.x
+        let newWidth = filterButtonWidthConstraint.constant - offset
+
+        if newWidth > maxFilterButtonWidth {
+            filterButtonWidthConstraint.constant = maxFilterButtonWidth
+        } else if newWidth < minFilterButtonWidth {
+            filterButtonWidthConstraint.constant = minFilterButtonWidth
+        } else {
+            filterButtonWidthConstraint.constant = newWidth
+            scrollView.contentOffset.x = 0
+        }
+
+        let sizeRange = maxFilterButtonWidth - minFilterButtonWidth
+        let visibleRatio = (filterButtonWidthConstraint.constant - minFilterButtonWidth) / sizeRange
+        let newAlpha: CGFloat = visibleRatio > 0.25 ? visibleRatio : 0
+        filterButtonView.updateLabel(withAlpha: newAlpha)
+    }
+}
+
+// MARK: - SearchFilterButtonViewDelegate
+
+extension SearchFilterTagsView: SearchFilterButtonViewDelegate {
+    func searchFilterButtonViewDidSelectFilter(_ searchFilterButtonView: SearchFilterButtonView) {
+        delegate?.searchFilterTagsViewDidSelectFilter(self)
+    }
+}
+
+// MARK: - SearchFilterTagCellDelegate
+
+extension SearchFilterTagsView: SearchFilterTagCellDelegate {
+    func searchFilterTagCellDidSelectRemove(_ cell: SearchFilterTagCell) {
+        guard let indexPath = collectionView.indexPath(for: cell) else {
+            return
+        }
+
+        UIImpactFeedbackGenerator(style: .medium).impactOccurred()
+
+        searchFilterTags.remove(at: indexPath.item)
+        collectionView.deleteItems(at: [indexPath])
+
+        delegate?.searchFilterTagsView(self, didRemoveTagAt: indexPath.item)
+    }
+
+    func searchFilterTagCellWasSelected(_ cell: SearchFilterTagCell) {
+        guard let indexPath = collectionView.indexPath(for: cell) else {
+            return
+        }
+        delegate?.searchFilterTagsView(self, didTapSearchFilterTagAt: indexPath.item)
+    }
+}
+
+// MARK: - UICollectionViewDelegateFlowLayout
+
+extension SearchFilterTagsView: UICollectionViewDelegateFlowLayout {
+    public func collectionView(_ collectionView: UICollectionView,
+                               layout collectionViewLayout: UICollectionViewLayout,
+                               sizeForItemAt indexPath: IndexPath) -> CGSize {
+
+        var cellWidth = SearchFilterTagCell.width(for: title(at: indexPath))
+        cellWidth = min(collectionView.bounds.width, cellWidth)
+        cellWidth = max(cellWidth, SearchFilterTagCell.minWidth)
+        return CGSize(width: cellWidth, height: SearchFilterTagCell.height)
+    }
+}
+
+// MARK: - Private extensions
+
+private extension SearchFilterTagsView {
+    private func searchFilterTagsAreEqual(_ lhs: [SearchFilterTagCellViewModel], _ rhs: [SearchFilterTagCellViewModel]) -> Bool {
```

Comment:
I'm not sure if it makes sense/is possible, but could `SearchFilterTagCellViewModel` implement `Equatable`? If so, we could remove this extension.

Reply:
Thanks for the feedback! 😁  I fixed all of them. 

The ViewModel was a protocol, but I changed it to a struct instead so it could implement `Equatable`. Then I could remove the extension. Looks much cleaner now ✨